packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,341 @@ import {
-        onClick={this._onCellClick}
-        onHover={this._onGridCellHovered}
-        onFocus={this._onGridCellFocused}
-        selected={this.state.selectedIndex !== undefined && this.state.selectedIndex === item.index}
-        circle={props.cellShape === 'circle'}
-        label={item.label}
-        onMouseEnter={this._onMouseEnter}
-        onMouseMove={this._onMouseMove}
-        onMouseLeave={this._onMouseLeave}
-        onWheel={this._onWheel}
-        onKeyDown={this._onKeyDown}
-        height={props.cellHeight}
-        width={props.cellWidth}
-        borderWidth={props.cellBorderWidth}
-      />
-    );
-  };
+  }, [props, internalState]);
 
   /**
    * Callback passed to the GridCell that will manage triggering the onCellHovered callback for mouseEnter
    */
-  private _onMouseEnter = (ev: React.MouseEvent<HTMLButtonElement>): boolean => {
-    if (!this.props.focusOnHover) {
-      return !this.isNavigationIdle || !!this.props.disabled;
-    }
-
-    if (this.isNavigationIdle && !this.props.disabled) {
-      ev.currentTarget.focus();
-    }
-
-    return true;
-  };
+  const onMouseEnter = React.useCallback(
+    (ev: React.MouseEvent<HTMLButtonElement>): boolean => {
+      if (!props.focusOnHover) {
+        return !internalState.isNavigationIdle || !!props.disabled;
+      }
+      if (internalState.isNavigationIdle && !props.disabled) {
+        ev.currentTarget.focus();
+      }
+      return true;
+    },
+    [props.focusOnHover, internalState.isNavigationIdle, props.disabled],
+  );
 
   /**
    * Callback passed to the GridCell that will manage Hover/Focus updates
    */
-  private _onMouseMove = (ev: React.MouseEvent<HTMLButtonElement>): boolean => {
-    if (!this.props.focusOnHover) {
-      return !this.isNavigationIdle || !!this.props.disabled;
-    }
+  const onMouseMove = React.useCallback(
+    (ev: React.MouseEvent<HTMLButtonElement>): boolean => {
+      if (!props.focusOnHover) {
+        return !internalState.isNavigationIdle || !!props.disabled;
+      }
 
-    const targetElement = ev.currentTarget as HTMLElement;
+      const targetElement = ev.currentTarget as HTMLElement;
 
-    // If navigation is idle and the targetElement is the focused element bail out
-    // if (!this.isNavigationIdle || (document && targetElement === (document.activeElement as HTMLElement))) {
-    if (this.isNavigationIdle && !(document && targetElement === (document.activeElement as HTMLElement))) {
-      targetElement.focus();
-    }
+      // If navigation is idle and the targetElement is the focused element bail out
+      // if (!this.isNavigationIdle || (document && targetElement === (document.activeElement as HTMLElement))) {
+      if (internalState.isNavigationIdle && !(document && targetElement === (document.activeElement as HTMLElement))) {
+        targetElement.focus();
+      }
 
-    return true;
-  };
+      return true;
+    },
+    [props.focusOnHover, internalState.isNavigationIdle, props.disabled],
```

Comment:
Similar comment about deps
```suggestion
    [focusOnHover, internalState, disabled],
```